tests/src/test/scala/runtime/actionContainers/PythonAdvancedTests.scala

```
@@ -91,4 +92,61 @@ trait PythonAdvancedTests {
         e shouldBe empty
     })
   }
+
+  Map(
+    "prelaunched" -> Map.empty[String, String],
+    "non-prelaunched" -> Map("OW_INIT_IN_ACTIONLOOP" -> ""),
+  ).foreach { case (name, env) =>
+    it should s"support a function with a lambda-like signature and transform HTTP events $name" in {
+      val (out, err) = withActionContainer(env) { c =>
+        val code =
+          """
+            |def main(event, context):
+            |   return {
+            |      "event": event,
+            |      "context": {
+            |         "remaining_time": context.get_remaining_time_in_millis(),
+            |         "activation_id": context.activation_id,
+            |         "function_name": context.function_name,
+            |         "function_version": context.function_version
+            |      }
+            |   }
+          """.stripMargin
+
+        val (initCode, _) = c.init(initPayload(code))
+        initCode should be(200)
+
+        val (runCode, out) = c.run(runPayload(
+          JsObject(
+            "__ow_headers" -> JsObject("headerKey" -> "headerValue".toJson),
+            "__ow_method" -> "get".toJson,
+            "__ow_path" -> "/foo/bar".toJson,
+            "__ow_query" -> "foo=bar&foo=baz&test=test".toJson
+          ), 
+          Some(JsObject(
+            "deadline" -> "0".toJson,
```

deadline = 0 could hide an error - how about testing with > 0 and ensuring remaining_time below is > 0 and <= n

---

True. I'll think about it more.